database/tbcd/level/level.go

```diff
@@ -989,17 +995,33 @@ func (l *ldb) BlockTxUpdate(ctx context.Context, direction int, txs map[tbcd.TxK
 			key = k[0:65]
 			value = nil
 
+			// insert block hash to determine if it was indexed later
+			if _, ok := bm[string(k[33:65])]; !ok {
+				copy(block[1:], k[33:65])
```

Comment:
I think we could avoid reslicing `key` twice by adding a variable:
```suggestion
			hash := k[33:65]
			if _, ok := bm[string(hash)]; !ok {
				copy(block[1:], hash)
```

Reply:
More often than not it will collide and then we make a variable that has single use. Also, that is cached in the tight loop.